### DISCUSSION TITLE
git clone error

### DISCUSSION THREAD

**Discussion by NicPhilipps** [community/community#68565] [New to GitHub]
I am trying to clone a directory onto my desktop and after entering my username and password from github I receive the following error: "Password for 'https://NicPhilipps@github.com': 
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/cubstart-ios-student/hw2-NicPhilipps.git/'
nicphilipps@wifi-10-40-82-113 ~ % "

Can anyone help me out how to solve this?

**Comment by seyLu** [comment]
#### Hi @NicPhilipps,

As the error message stated, Github officially phased out passwords way back on Aug 31, 2021. If you have your own machine, it is better to use SSH to authenticate to Github (since tokens expire by default and SSH don't).

You can read more on:
- [Github blog: Token requirements for git operations](https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/)
- [Github docs: About SSH](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/about-ssh)
- [Github docs: Test for existing SSH keys](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/checking-for-existing-ssh-keys)
- [Github docs: Generating a new SSH key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)
- [Github docs: Adding a new SSH key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)
- [Github docs: Testing SSH connection](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/testing-your-ssh-connection)

**Comment by sreshtech** [comment]
The error message you received is because GitHub no longer supports password authentication for cloning repositories over HTTPS . Instead, they recommend using a personal access token or SSH key for authentication. Here's a simple step-by-step guide on how to resolve this issue:

Step 1: Create a Personal Access Token
1. Go to the GitHub website and log in to your account if you haven't already.

2. Click on your profile picture in the upper right corner and select "Settings."

3. In the left sidebar, click on "Developer settings."

4. On the left sidebar under "Access tokens," click "Personal access tokens."

5. Click the "Generate token" button.

6. Give your token a name, select the desired expiration, and choose the scopes (permissions) you need. For cloning repositories, you should at least select "repo" and "read:user."

7. Scroll down and click the "Generate token" button at the bottom of the page.

8. GitHub will generate a personal access token for you. >>> Copy and paste this token somewhere safe because you won't be able to see it again.

Step 2: Use the Personal Access Token to Clone the Repository

Now that you have your personal access token, you can use it to clone your GitHub repository onto your desktop. Open your terminal and run the following command, replacing `<your_personal_access_token>` with the token you generated in Step 1:

bash
git clone https://<your_personal_access_token>@github.com/cubstart-ios-student/hw2-NicPhilipps.git


So, the full command might look something like this:

=>bash
git clone https://your-access-token@github.com/cubstart-ios-student/hw2-NicPhilipps.git

After entering this command, you shouldn't be prompted for a password, and the repository should clone successfully.

This approach ensures secure and authenticated access to your GitHub repositories while following GitHub's new authentication guidelines.

**Comment by shivangvijay** [comment]
I suggest to create access token and kept that somewhere saved in your laptop. Steps are:-

Access Personal Settings:

Click on your profile picture in the upper-right corner of the GitHub interface, and select "Settings" from the dropdown menu.

Navigate to Developer Settings:

In the left sidebar, scroll down and click on "Developer settings."

Access Personal Access Tokens:

In the "Developer settings" section, click on "Personal access tokens" under the "Access tokens" category.

Generate a New Token:

 Click the "Generate token" button.

Provide a Token Name:

Enter a name for the token that helps identify its purpose or usage. For example, "Access for Repository XYZ."

Select Scopes:

GitHub allows you to select specific scopes (permissions) for the token. Select the scopes that match the level of access required. Be cautious and only grant the necessary permissions.

Generate Token:

 After selecting the appropriate scopes, scroll down and click the "Generate token" button.

Copy the Token:

 Once generated, a personal access token will be displayed on the screen. It's important to copy the token immediately as it will not be shown again for security reasons. Store it securely.
